Automattic/wp-calypso
# Plans: Adjust product ordering to match design mockups

Fixes `1196341175636977-as-1198202606132769`.

#### Changes proposed in this Pull Request

* Add an explicit ordering for products and plans in `ProductsGridAlt` to match design mock-ups.
* Separate rendering of the current/legacy plan card from the process of showing the other product cards.
* Remove lots of old, unneeded code imported from the original `PlansColumn` and `ProductsColumn`.

#### Testing instructions

* Open the Plans page (any context or environment will do: Calypso, Connect, or Cloud).
* Verify that products and plans are listed in the following order, for both monthly and yearly durations:
  1. Your existing "legacy" plan, e.g., Premium (only if your site has one)
  2. Jetpack Security
  3. Jetpack Backup
  4. Jetpack Scan
  5. Jetpack Search
  6. Jetpack CRM
  7. Jetpack Anti-spam
  8. Jetpack Complete
  9. Jetpack Free (only in Connect and Cloud)
* Verify that clicking the call-to-action button on each card goes to the correct place for that card and the selected duration, as indicated by the button text.
* Verify that if you already own a product with options (i.e., Jetpack Security and/or Jetpack Backup), those cards correctly display which option you own (i.e., Daily or Real-time).
* Verify that if you own a monthly product or plan, when you switch to the Yearly duration option, you see a call-to-action button that says "Upgrade to Yearly."

#### Reference screenshots

##### Active legacy plan

<img width="571" alt="Screen Shot 2020-10-13 at 09 35 53" src="https://user-images.githubusercontent.com/670067/95878634-0ef37580-0d3b-11eb-9352-19375def5f45.png">

##### No active legacy plan

<img width="569" alt="image" src="https://user-images.githubusercontent.com/670067/95879143-9fca5100-0d3b-11eb-8e30-7b5d0a357120.png">
